include/integrations/cache/cache.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/*************************************************************************************************
+ * Copyright 2017 JPL TSolucio, S.L. -- This file is a part of TSOLUCIO coreBOS customizations.
+ * You can copy, adapt and distribute the work under the "Attribution-NonCommercial-ShareAlike"
+ * Vizsage Public License (the "License"). You may not use this file except in compliance with the
+ * License. Roughly speaking, non-commercial users may share and modify this code, but must give credit
+ * and share improvements. However, for proper details please read the full License, available at
+ * http://vizsage.com/license/Vizsage-License-BY-NC-SA.html and the handy reference for understanding
+ * the full license at http://vizsage.com/license/Vizsage-Deed-BY-NC-SA.html. Unless required by
+ * applicable law or agreed to in writing, any software distributed under the License is distributed
+ * on an  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the
+ * License terms of Creative Commons Attribution-NonCommercial-ShareAlike 3.0 (the License).
+ * ************************************************************************************************
+ *  Module    : Whatsapp Integration
+ *  Version   : 1.0
+ *  Author    : JPL TSolucio, S. L.
+ *************************************************************************************************/
+require_once 'vtlib/Vtiger/Module.php';
+require "vendor/autoload.php";
+
+// Use the REST API Client to make requests to the Twilio REST API
+use Twilio\Rest\Client;
+
+class corebos_cache {
+
+	// Configuration Properties
+	private $adapter;
+	private $ip;
+	private $port;
+
+	// Configuration Keys
+	const KEY_ISACTIVE = 'cache_isactive';
+	const KEY_ADAPTER = 'cache_adapter';
+	const KEY_IP = 'cache_ip';
+	const KEY_PORT = 'cache_port';
+
+	const ADAPTER_MEMORY = "memory";
+	const ADAPTER_REDIS = "redis";
+	const ADAPTER_MEMCACHED = "memcached";
+
+	// Utilities
+	private static $cacheClient = null;
+
+	public function __construct() {
+		$this->initGlobalScope();
+	}
+
+	public function initGlobalScope() {
+		$this->adapter = coreBOS_Settings::getSetting(self::KEY_ADAPTER, '');
+		$this->ip = coreBOS_Settings::getSetting(self::KEY_IP, '');
+		$this->port = coreBOS_Settings::getSetting(self::KEY_PORT, '');
+		if ($this->adapter == self::ADAPTER_MEMORY) {
```

Comment:
what happens here when $this->adapter!=ADAPTER_MEMORY and the IP or PORT are empty? how does this initialize?